amazon/ionbenchmark/Format.py

```diff
@@ -19,7 +36,37 @@ def format_is_binary(format_option):
 
 
 def rewrite_file_to_format(file, format_option):
-    return file
+    temp_file_name_base = 'temp_' + os.path.splitext(os.path.basename(file))[0]
+    if format_option == Format.ION_BINARY.value:
+        temp_file_name_suffix = '.10n'
+    elif format_option == Format.ION_TEXT.value:
+        temp_file_name_suffix = '.ion'
+    else:
+        temp_file_name_suffix = ''
+    temp_file_name = temp_file_name_base + temp_file_name_suffix
+    # Check the file path
+    if os.path.exists(temp_file_name):
+        os.remove(temp_file_name)
+
+    if format_is_ion(format_option):
+        # Write data if a conversion is required
+        if (format_option == Format.ION_BINARY.value and file_is_ion_text(file)) \
+                or (format_option == Format.ION_TEXT.value and file_is_ion_binary(file)):
+            # Load data
+            with open(file, 'br') as fp:
+                obj = simpleion.load(fp, single_value=False)
+            with open(temp_file_name, 'bw') as fp:
+                if format_option == Format.ION_BINARY.value:
+                    simpleion.dump(obj, fp, binary=True)
+                else:
+                    simpleion.dump(obj, fp, binary=False)
+        else:
+            shutil.copy(file, temp_file_name)
+    else:
+        # Copy the file
+        shutil.copy(file, temp_file_name)
```

Comment:
For large files, always copying may become a burden, if no conversion was required.

Reply:
- https://github.com/amazon-ion/ion-python/issues/269